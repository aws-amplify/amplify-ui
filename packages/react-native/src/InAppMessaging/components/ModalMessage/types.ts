import { ModalMessageCommonProps } from '@aws-amplify/ui-react-core-notifications';

import { MessageDefaultStyle, MessageOverrideStyle } from '../../hooks';

export interface ModalMessageProps
  extends ModalMessageCommonProps<MessageOverrideStyle | undefined> {}

export interface ModalMessageStyle extends MessageDefaultStyle {}
