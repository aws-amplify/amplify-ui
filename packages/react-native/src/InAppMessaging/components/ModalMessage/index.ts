export { default as ModalMessage } from './ModalMessage';
export { ModalMessageProps } from './types';
