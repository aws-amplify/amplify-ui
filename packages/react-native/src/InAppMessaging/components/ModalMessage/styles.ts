import { ImageStyle, StyleSheet } from 'react-native';
import {
  BORDER_RADIUS_BASE,
  COLOR_BLACK,
  COLOR_LIGHT_GREY,
  COLOR_WHITE,
  FONT_SIZE_BASE,
  FONT_SIZE_LARGE,
  FONT_WEIGHT_BASE,
  FONT_WEIGHT_BOLD,
  LINE_HEIGHT_BASE,
  LINE_HEIGHT_LARGE,
  MESSAGE_ELEVATION,
  MESSAGE_SHADOW_HEIGHT,
  MESSAGE_SHADOW_OPACITY,
  MESSAGE_SHADOW_RADIUS,
  MESSAGE_SHADOW_WIDTH,
  SPACING_EXTRA_LARGE,
  SPACING_LARGE,
  SPACING_MEDIUM,
  SPACING_SMALL,
} from '../../constants';

import { ModalMessageStyle } from './types';

const commonStyles: Omit<
  ModalMessageStyle,
  'contentContainer' | 'image' | 'textContainer'
> = {
  body: {
    fontSize: FONT_SIZE_BASE,
    fontWeight: FONT_WEIGHT_BASE,
    lineHeight: LINE_HEIGHT_BASE,
  },
  buttonContainer: {
    backgroundColor: COLOR_LIGHT_GREY,
    borderRadius: BORDER_RADIUS_BASE,
    flex: 1,
    margin: SPACING_MEDIUM,
    padding: SPACING_LARGE,
  },
  buttonsContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
  container: {
    backgroundColor: COLOR_WHITE,
    elevation: MESSAGE_ELEVATION,
    margin: SPACING_EXTRA_LARGE,
    shadowColor: COLOR_BLACK,
    shadowOffset: {
      width: MESSAGE_SHADOW_WIDTH,
      height: MESSAGE_SHADOW_HEIGHT,
    },
    shadowOpacity: MESSAGE_SHADOW_OPACITY,
    shadowRadius: MESSAGE_SHADOW_RADIUS,
  },
  buttonText: {
    fontSize: FONT_SIZE_BASE,
    fontWeight: FONT_WEIGHT_BASE,
    lineHeight: LINE_HEIGHT_BASE,
    textAlign: 'center',
  },
  header: {
    fontSize: FONT_SIZE_LARGE,
    fontWeight: FONT_WEIGHT_BOLD,
    lineHeight: LINE_HEIGHT_LARGE,
  },
  iconButton: {
    alignSelf: 'flex-start',
    marginLeft: 'auto',
  },
  imageContainer: {
    alignItems: 'center',
    margin: SPACING_LARGE,
  },
  wrapper: {
    flex: 1,
    justifyContent: 'center',
  },
};

export const getPortraitStyles = (
  imageDimensions: ImageStyle
): ModalMessageStyle =>
  StyleSheet.create({
    ...commonStyles,
    buttonsContainer: {
      ...commonStyles.buttonsContainer,
      marginTop: 'auto',
      paddingHorizontal: SPACING_SMALL,
      paddingBottom: SPACING_SMALL,
    },
    container: {
      ...commonStyles.container,
      minHeight: '40%',
    },
    contentContainer: { padding: SPACING_LARGE },
    image: { ...imageDimensions },
    textContainer: {
      marginTop: SPACING_LARGE,
    },
  });

export const getLandscapeStyles = (
  imageDimensions: ImageStyle
): ModalMessageStyle =>
  StyleSheet.create({
    ...commonStyles,
    container: {
      ...commonStyles.container,
      flex: 1,
      padding: SPACING_EXTRA_LARGE,
    },
    contentContainer: {
      flex: 1,
      flexDirection: 'row',
    },
    iconButton: {
      ...commonStyles.iconButton,
      marginRight: SPACING_MEDIUM,
    },
    image: { ...imageDimensions },
    imageContainer: {
      ...commonStyles.imageContainer,
      justifyContent: 'center',
    },
    textContainer: {
      flex: 1,
      justifyContent: 'center',
      paddingHorizontal: SPACING_MEDIUM,
    },
  });
