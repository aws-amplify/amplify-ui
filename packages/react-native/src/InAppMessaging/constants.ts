import { getLineHeight } from '../utils';

// color
export const COLOR_BLACK = '#000';
export const COLOR_GREY = '#a1a1a1';
export const COLOR_LIGHT_GREY = '#e8e8e8';
export const COLOR_LIGHT_GREY_2 = '#d8d8d8';
export const COLOR_WHITE = '#fff';

// spacing
export const SPACING_SMALL = 4;
export const SPACING_MEDIUM = 8;
export const SPACING_LARGE = 12;
export const SPACING_EXTRA_LARGE = 16;

// border radius
export const BORDER_RADIUS_BASE = 4;

// font
export const FONT_SIZE_BASE = 16;
export const FONT_SIZE_LARGE = 18;

export const LINE_HEIGHT_BASE = getLineHeight(FONT_SIZE_BASE);
export const LINE_HEIGHT_LARGE = getLineHeight(FONT_SIZE_LARGE);

export const FONT_WEIGHT_BASE = '400';
export const FONT_WEIGHT_BOLD = '600';

// icon
export const ICON_BUTTON_SIZE = 20;
export const ICON_BUTTON_HIT_SLOP = 10;

// shadow properties

// iOS shadow values
export const MESSAGE_SHADOW_HEIGHT = 2;
export const MESSAGE_SHADOW_WIDTH = 2;
export const MESSAGE_SHADOW_OPACITY = 0.1;
export const MESSAGE_SHADOW_RADIUS = 2;

// android shadow values
export const MESSAGE_ELEVATION = 3;

// Message UI Buttons

// default value applied in React Native TouchableOpacity
export const BUTTON_PRESSED_OPACITY = 0.8;

// Message Component Test IDs
export const IN_APP_MESSAGING_TEST_ID = {
  BODY: 'in-app-messaging--body',
  CLOSE_BUTTON: 'in-app-messaging--close-button',
  HEADER: 'in-app-messaging--header',
  IMAGE: 'in-app-messaging--image',
  PRIMARY_BUTTON: 'in-app-messaging--primary-button',
  SECONDARY_BUTTON: 'in-app-messaging--secondary-button',
  CAROUSEL: 'in-app-messaging--carousel',
};
