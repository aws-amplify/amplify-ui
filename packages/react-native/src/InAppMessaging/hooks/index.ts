export { ImageDimensions, useMessageImage } from './useMessageImage';
export {
  GetDefaultStyle,
  MessageComponentStyles,
  MessageDefaultStyle,
  MessageOverrideStyle,
  useMessageProps,
  UseMessageProps,
} from './useMessageProps';
