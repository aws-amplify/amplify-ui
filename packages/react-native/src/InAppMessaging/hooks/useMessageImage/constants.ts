import { Dimensions } from 'react-native';

// as images are not expected to be responsive to orientation changes get screen dimensions at app start
const SCREEN_DIMENSIONS = Dimensions.get('screen');

// compare screen width and height and assign the lesser of the two as the base screen dimension
const BASE_SCREEN_DIMENSION =
  SCREEN_DIMENSIONS.width < SCREEN_DIMENSIONS.height
    ? SCREEN_DIMENSIONS.width
    : SCREEN_DIMENSIONS.height;

// base size that message images should fill
// - all banner message images should fill 20 percent of the base screen dimension
// - all other components should fill 60 percent of the base screen dimension
export const BANNER_IMAGE_SCREEN_MULTIPLIER = 0.2;
export const CAROUSEL_IMAGE_SCREEN_MULTIPLIER = 0.6;
export const FULL_SCREEN_IMAGE_SCREEN_MULTIPLIER = 0.6;
export const MODAL_IMAGE_SCREEN_MULTIPLIER = 0.6;

export const BANNER_IMAGE_SCREEN_SIZE =
  BANNER_IMAGE_SCREEN_MULTIPLIER * BASE_SCREEN_DIMENSION;
export const CAROUSEL_IMAGE_SCREEN_SIZE =
  CAROUSEL_IMAGE_SCREEN_MULTIPLIER * BASE_SCREEN_DIMENSION;
export const FULL_SCREEN_IMAGE_SCREEN_SIZE =
  FULL_SCREEN_IMAGE_SCREEN_MULTIPLIER * BASE_SCREEN_DIMENSION;
export const MODAL_IMAGE_SCREEN_SIZE =
  MODAL_IMAGE_SCREEN_MULTIPLIER * BASE_SCREEN_DIMENSION;
