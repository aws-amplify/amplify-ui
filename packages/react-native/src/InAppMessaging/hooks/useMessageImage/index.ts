export { default as useMessageImage } from './useMessageImage';
export { ImageDimensions } from './types';
