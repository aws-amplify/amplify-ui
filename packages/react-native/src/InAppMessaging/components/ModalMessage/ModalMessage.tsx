import React from 'react';

import { useDeviceOrientation } from '../../../hooks';
import { useMessageProps } from '../../hooks';
import { MessageLayout } from '../MessageLayout';
import { MessageWrapper } from '../MessageWrapper';

import { getLandscapeStyles, getPortraitStyles } from './styles';
import { ModalMessageProps } from './types';

export default function ModalMessage(
  props: ModalMessageProps
): JSX.Element | null {
  const { deviceOrientation, isPortraitMode } = useDeviceOrientation();
  const messageProps = useMessageProps(
    props,
    isPortraitMode ? getPortraitStyles : getLandscapeStyles
  );
  const { shouldRenderMessage, styles } = messageProps;

  if (!shouldRenderMessage) {
    return null;
  }

  const { wrapper, ...messageStyles } = styles!;

  return (
    <MessageWrapper style={wrapper}>
      <MessageLayout
        {...props}
        {...messageProps}
        orientation={deviceOrientation}
        styles={messageStyles}
        testID={`inappmessaging-modal-dialog`}
      />
    </MessageWrapper>
  );
}
