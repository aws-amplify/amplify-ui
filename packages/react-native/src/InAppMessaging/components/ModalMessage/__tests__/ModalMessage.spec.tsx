import React from 'react';
import TestRenderer from 'react-test-renderer';

import { IN_APP_MESSAGING_TEST_ID } from '../../../constants';
import { useDeviceOrientation } from '../../../../hooks';
import { useMessageImage } from '../../../hooks';

import ModalMessage from '../ModalMessage';

jest.mock('../../../../hooks');
jest.mock('../../../hooks/useMessageImage');
jest.mock('../../MessageWrapper', () => ({ MessageWrapper: 'MessageWrapper' }));

const mockUseMessageImage = useMessageImage as jest.Mock;
const onAction = jest.fn();
const onClose = jest.fn();

const baseProps = { layout: 'MODAL' as const, onClose };

describe('ModalMessage', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it.each([
    ['landscape', false],
    ['portrait', true],
  ])('renders as expected in %s mode', (deviceOrientation, isPortraitMode) => {
    (useDeviceOrientation as jest.Mock).mockReturnValue({
      deviceOrientation,
      isPortraitMode,
    });
    mockUseMessageImage.mockReturnValueOnce({
      hasRenderableImage: false,
      imageDimensions: { height: null, width: null },
      isImageFetching: false,
    });

    const renderer = TestRenderer.create(<ModalMessage {...baseProps} />);

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('renders a message as expected with an image', () => {
    (useDeviceOrientation as jest.Mock).mockReturnValue({
      deviceOrientation: 'portrait',
      isPortraitMode: true,
    });
    mockUseMessageImage.mockReturnValueOnce({
      hasRenderableImage: true,
      imageDimensions: { height: 100, width: 100 },
      isImageFetching: false,
    });

    const src = 'asset.png';
    const props = { ...baseProps, image: { src } };

    const renderer = TestRenderer.create(<ModalMessage {...props} />);

    const image = renderer.root.findByProps({
      testID: IN_APP_MESSAGING_TEST_ID.IMAGE,
    });

    expect(image.props).toEqual(
      expect.objectContaining({ source: { uri: src } })
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('returns null while an image is fetching', () => {
    (useDeviceOrientation as jest.Mock).mockReturnValue({
      deviceOrientation: 'portrait',
      isPortraitMode: true,
    });
    mockUseMessageImage.mockReturnValueOnce({
      hasRenderableImage: false,
      imageDimensions: { height: null, width: null },
      isImageFetching: true,
    });

    const renderer = TestRenderer.create(<ModalMessage {...baseProps} />);

    expect(renderer.toJSON()).toBeNull();
  });

  it.each([
    [
      'header',
      IN_APP_MESSAGING_TEST_ID.HEADER,
      {
        testProps: { content: 'header content' },
        expectedProps: { children: 'header content' },
      },
    ],
    [
      'body',
      IN_APP_MESSAGING_TEST_ID.BODY,
      {
        testProps: { content: 'body content' },
        expectedProps: { children: 'body content' },
      },
    ],
    [
      'primaryButton',
      IN_APP_MESSAGING_TEST_ID.PRIMARY_BUTTON,
      {
        testProps: { onAction, title: 'primary button' },
        expectedProps: { children: 'primary button', onPress: onAction },
      },
    ],
    [
      'secondaryButton',
      IN_APP_MESSAGING_TEST_ID.SECONDARY_BUTTON,
      {
        testProps: { onAction, title: 'secondary button' },
        expectedProps: { children: 'secondary button', onPress: onAction },
      },
    ],
  ])(
    'correctly handles a %s prop',
    (key, testID, { testProps, expectedProps }) => {
      mockUseMessageImage.mockReturnValueOnce({
        hasRenderableImage: false,
        imageDimensions: { height: null, width: null },
        isImageFetching: false,
      });

      const props = { ...baseProps, [key]: testProps };

      const renderer = TestRenderer.create(<ModalMessage {...props} />);
      const testElement = renderer.root.findByProps({ testID });

      expect(testElement.props).toEqual(expect.objectContaining(expectedProps));
    }
  );

  it('calls onClose when the close button is pressed', () => {
    mockUseMessageImage.mockReturnValueOnce({
      hasRenderableImage: false,
      imageDimensions: { height: null, width: null },
      isImageFetching: false,
    });

    const renderer = TestRenderer.create(<ModalMessage {...baseProps} />);
    const closeButton = renderer.root.findByProps({
      testID: IN_APP_MESSAGING_TEST_ID.CLOSE_BUTTON,
    });

    TestRenderer.act(() => {
      (closeButton.props as { onPress: () => void }).onPress();
    });

    expect(onClose).toHaveBeenCalledTimes(1);
  });
});
