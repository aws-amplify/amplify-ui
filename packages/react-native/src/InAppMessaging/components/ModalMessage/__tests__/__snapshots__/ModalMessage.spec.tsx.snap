// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalMessage correctly handles a primaryButton prop 1`] = `
<MessageWrapper
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "elevation": 3,
          "margin": 16,
          "minHeight": "40%",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 2,
        },
        Object {},
        Object {},
      ]
    }
    testID="inappmessaging-modal-dialog"
  >
    <View
      style={
        Object {
          "padding": 12,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {},
            undefined,
            Array [
              Object {
                "alignSelf": "flex-start",
                "marginLeft": "auto",
              },
              undefined,
            ],
          ]
        }
        testID="in-app-messaging--close-button"
      >
        <Image
          accessibilityRole="image"
          source={
            Object {
              "testUri": "../../../src/assets/icons/close.png",
            }
          }
          style={
            Array [
              Object {
                "height": 20,
                "resizeMode": "contain",
                "tintColor": undefined,
                "width": 20,
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": "auto",
          "paddingBottom": 4,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "hsl(210, 10%, 58%)",
              "borderRadius": 4,
              "borderWidth": 1,
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            },
            undefined,
            Array [
              Object {},
              Object {
                "backgroundColor": "#e8e8e8",
                "borderRadius": 4,
                "flex": 1,
                "margin": 8,
                "padding": 12,
              },
              Object {},
              Object {},
            ],
          ]
        }
        testID="in-app-messaging--primary-button"
      >
        <Text
          style={
            Array [
              Object {
                "color": "hsl(210, 50%, 10%)",
                "fontSize": 16,
                "fontWeight": "700",
                "textAlign": "center",
              },
              Array [
                Object {
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 24,
                  "textAlign": "center",
                },
                Object {},
                Object {},
              ],
            ]
          }
        >
          primary button
        </Text>
      </View>
    </View>
  </View>
</MessageWrapper>
`;

exports[`ModalMessage correctly handles a secondaryButton prop 1`] = `
<MessageWrapper
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "elevation": 3,
          "margin": 16,
          "minHeight": "40%",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 2,
        },
        Object {},
        Object {},
      ]
    }
    testID="inappmessaging-modal-dialog"
  >
    <View
      style={
        Object {
          "padding": 12,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {},
            undefined,
            Array [
              Object {
                "alignSelf": "flex-start",
                "marginLeft": "auto",
              },
              undefined,
            ],
          ]
        }
        testID="in-app-messaging--close-button"
      >
        <Image
          accessibilityRole="image"
          source={
            Object {
              "testUri": "../../../src/assets/icons/close.png",
            }
          }
          style={
            Array [
              Object {
                "height": 20,
                "resizeMode": "contain",
                "tintColor": undefined,
                "width": 20,
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": "auto",
          "paddingBottom": 4,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "hsl(210, 10%, 58%)",
              "borderRadius": 4,
              "borderWidth": 1,
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            },
            undefined,
            Array [
              Object {},
              Object {
                "backgroundColor": "#e8e8e8",
                "borderRadius": 4,
                "flex": 1,
                "margin": 8,
                "padding": 12,
              },
              Object {},
              Object {},
            ],
          ]
        }
        testID="in-app-messaging--secondary-button"
      >
        <Text
          style={
            Array [
              Object {
                "color": "hsl(210, 50%, 10%)",
                "fontSize": 16,
                "fontWeight": "700",
                "textAlign": "center",
              },
              Array [
                Object {
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 24,
                  "textAlign": "center",
                },
                Object {},
                Object {},
              ],
            ]
          }
        >
          secondary button
        </Text>
      </View>
    </View>
  </View>
</MessageWrapper>
`;

exports[`ModalMessage renders a message as expected with an image 1`] = `
<MessageWrapper
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "elevation": 3,
          "margin": 16,
          "minHeight": "40%",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 2,
        },
        Object {},
        Object {},
      ]
    }
    testID="inappmessaging-modal-dialog"
  >
    <View
      style={
        Object {
          "padding": 12,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {},
            undefined,
            Array [
              Object {
                "alignSelf": "flex-start",
                "marginLeft": "auto",
              },
              undefined,
            ],
          ]
        }
        testID="in-app-messaging--close-button"
      >
        <Image
          accessibilityRole="image"
          source={
            Object {
              "testUri": "../../../src/assets/icons/close.png",
            }
          }
          style={
            Array [
              Object {
                "height": 20,
                "resizeMode": "contain",
                "tintColor": undefined,
                "width": 20,
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "margin": 12,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "asset.png",
            }
          }
          style={
            Array [
              Object {
                "height": 100,
                "width": 100,
              },
              undefined,
            ]
          }
          testID="in-app-messaging--image"
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
    </View>
  </View>
</MessageWrapper>
`;

exports[`ModalMessage renders as expected in landscape mode 1`] = `
<MessageWrapper
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "elevation": 3,
          "flex": 1,
          "margin": 16,
          "padding": 16,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 2,
        },
        Object {},
        Object {},
      ]
    }
    testID="inappmessaging-modal-dialog"
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "paddingHorizontal": 8,
          }
        }
      />
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {},
            undefined,
            Array [
              Object {
                "alignSelf": "flex-start",
                "marginLeft": "auto",
                "marginRight": 8,
              },
              undefined,
            ],
          ]
        }
        testID="in-app-messaging--close-button"
      >
        <Image
          accessibilityRole="image"
          source={
            Object {
              "testUri": "../../../src/assets/icons/close.png",
            }
          }
          style={
            Array [
              Object {
                "height": 20,
                "resizeMode": "contain",
                "tintColor": undefined,
                "width": 20,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</MessageWrapper>
`;

exports[`ModalMessage renders as expected in portrait mode 1`] = `
<MessageWrapper
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "elevation": 3,
          "margin": 16,
          "minHeight": "40%",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 2,
        },
        Object {},
        Object {},
      ]
    }
    testID="inappmessaging-modal-dialog"
  >
    <View
      style={
        Object {
          "padding": 12,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {},
            undefined,
            Array [
              Object {
                "alignSelf": "flex-start",
                "marginLeft": "auto",
              },
              undefined,
            ],
          ]
        }
        testID="in-app-messaging--close-button"
      >
        <Image
          accessibilityRole="image"
          source={
            Object {
              "testUri": "../../../src/assets/icons/close.png",
            }
          }
          style={
            Array [
              Object {
                "height": 20,
                "resizeMode": "contain",
                "tintColor": undefined,
                "width": 20,
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
    </View>
  </View>
</MessageWrapper>
`;
